src/datasets/arrow_dataset.py

```diff
@@ -2526,6 +2585,43 @@ def map(
             else:
                 return self
 
+        if self.info.splits:
+            splits = {}
+
+            def cache_file_name_for_split(cache_file_name: str, split_name: str) -> str:
+                sep = cache_file_name.rindex(".")
+                base_name, extension = cache_file_name[:sep], cache_file_name[sep:]
+                cache_file_name = base_name + f"-{split_name}" + extension
+                return cache_file_name
+
+            for split_name in self.info.splits:
+                splits[split_name] = self[split_name].map(
+                    function,
+                    with_indices=with_indices,
+                    with_rank=with_rank,
+                    input_columns=input_columns,
+                    batched=batched,
+                    batch_size=batch_size,
+                    drop_last_batch=drop_last_batch,
+                    remove_columns=remove_columns,
+                    keep_in_memory=keep_in_memory,
+                    load_from_cache_file=load_from_cache_file,
+                    cache_file_name=cache_file_name_for_split(cache_file_name, split_name)
+                    if cache_file_name is not None
+                    else None,
+                    writer_batch_size=writer_batch_size,
+                    features=features,
+                    disable_nullable=disable_nullable,
+                    fn_kwargs=fn_kwargs,
+                    num_proc=num_proc,
+                    suffix_template=suffix_template,
+                    new_fingerprint=new_fingerprint + f"-{split_name}",
```

Comment:
```suggestion
                    new_fingerprint= f"{new_fingerprint}-{split_name}" if new_fingerprint else None,
```
otherwise would raise `unsupported operand type(s) for +: 'NoneType' and 'str'`